test/e2e/framework/framework.go

```diff
@@ -247,3 +251,37 @@ func GetDeletionChannel(informer cache.SharedIndexInformer) chan string {
 
 	return deletionChannel
 }
+
+func (f *Framework) AwaitRouteAgentPodOnNode(cluster framework.ClusterIndex, nodeName string, prevPodUID types.UID) *corev1.Pod {
```

Comment:
Thanks for working on it @pinikomarov. 

Suggestion: We already have few APIs for GatewayPod which does similar functionality. Can we create wrappers to those APIs 

For example in Shipyard we can add the following API and use it from Submariner.

// AwaitSubmarinerRouteAgentPod finds the Route Agent pod in a given cluster, waiting if necessary for a period of time
// for the pod to materialize.
func (f *Framework) AwaitSubmarinerRouteAgentPod(cluster ClusterIndex) *v1.Pod {
	return &f.AwaitPodsByAppLabel(cluster, RouteAgent, TestContext.SubmarinerNamespace, 1).Items[0]
}

and another wrapper to check that its a new Pod (similar to what we have in the API AwaitNewSubmarinerEnginePod, file: gateway_failover.go)?

@tpantelis @mangelajo your thoughts?

Reply:
Actually I'm finishing up this PR. Originally @pinikomarov pushed a PR to add such a method to shipyard - we can do that but I thought it wasn't generally useful outside of submariner and route agent is specific to submariner.

Reply:
As Submariner Engine/Gateway falls in the same category, IMO it's better to keep the APIs together (either in Shipyard or in Submariner repo)

Reply:
https://github.com/submariner-io/shipyard/pull/440